todo/src/components/TodoList.js

```diff
@@ -0,0 +1,100 @@
+import React, {Component, Fragment} from 'react';
+import {connect} from 'react-redux';
+import { IoMdClose } from 'react-icons/io';
+
+import {markTaskComplete, clearCompleted, addTask, removeTask} from './actions'
+
+
+// Yes, yes, I know, this is one BIG component with ALL my redux crammed into it!
+// But, my argument here is, Redux is already a whole lot more typing. I can break these
+// into individual components, that's actually trivial at this point.
+// And I will definitely do that when we move into groups or on production sites.
+// But for this demo/learning tool, etc, it's MUCH simpler to just keep most functionality
+// in one file. Easier to reference, easier to see what's going on, and I think easier
+// to learn with. 
+class TodoList extends Component {
+    state = {
+        taskInput:''
+    };
+
+    markTaskComplete = id => {
+        this.props.markTaskComplete(id);
+    }
+
+    clearCompleted = () => {
+        this.props.clearCompleted();
+    }
+
+    handleSubmit = event => {
```

Comment:
handleSubmit and handleInput are really the only functions you would want to declare at the component level.